_posts/2023-02-06-tesla-ypp-2022-blogpost.md

```diff
@@ -0,0 +1,245 @@
+This to validate that the application can be build using maven.
+Another trigger for the CI/CD to run is when code is pushed to the develop branch.
+
+4. Java Spring Boot: Java Spring Boot is a framework used for building microservices and web applications.
+In our project, we utilized the power of Java Spring Boot to build the back-end of our application.
+Our project consisted of several different microservices, including the RPS proxy, RPS backend, stub Tesla API, Google application, and Tesla authenticator.
+Each of these microservices played a crucial role in the functionality and performance of our overall application.
+
+5. Google Actions: Google Actions is a platform for building conversational experiences for Google Assistant.
+We used Google Actions to build and integrate conversational interfaces into our application.
+For authenticating our users, google action is linked to our Cognito.
+Google actions send different request, like for instance a synchronization, to the webhook of our Google Application.
+
+6. Docker: Docker is a containerization technology used for deploying and managing applications.
+We used Docker to containerize our application, which allowed us to easily deploy and manage our application.
+
+7. Kubernetes + Helm: Kubernetes is a container orchestration platform used for automating the deployment, scaling, and management of containerized applications.
+We used Kubernetes to manage our Docker containers and Helm to simplify the deployment and management of our application.
+
+8. Renovate: Renovate is a tool used for automating the updating of dependencies in our application.
+We used Renovate to ensure that our dependencies were always up-to-date and secure.
+It automatically makes a pull request when a new version of a dependency is found.
+We’ve set a limit at 2 pull requests made by renovate, so that we can maintain a good overview.
+
+9. Postman: Postman is a tool used for testing and documenting APIs.
+We used Postman to test our APIs, which allowed us to ensure that our APIs were working correctly and that they were easy to use for other developers.
+
+10. Gatling: Gatling is a load testing tool used to test the performance of our application.
+We used Gatling to ensure that our application could handle high volumes of traffic and remain performant under heavy load.
+
+## Final Product
+
+Our final product consists of two applications designed to offer a unique and engaging user experience.
+The first application is a fun and interactive game allowing users to play the classic rock paper scissors with a tesla.
+
+The second application is a Google Home app that enables users to control a Tesla car using Google Assistant.
+With this application, users
+can quickly check their vehicle's temperature and battery level and start and stop charging with just a few simple voice commands.
+Users can connect Google Assistant to their Tesla car quickly and easily, offering a fun solution to manage their vehicle.
```

Comment:
```suggestion
Users can connect Google Assistant to their Tesla car quickly and easily, offering a easy to use way to manage their vehicle.
```

Reply:
The Google Assistant doesn't necessarily make it easier to use.

Reply:
The way to connect is easy, but the Google Assistant doesn't necessarily make the management of the car easier.